src/ClientData/include/ClientData/DataManager.h

```diff
@@ -29,6 +29,18 @@
 
 namespace orbit_client_data {
 
+// TimeRange represents an inclusive time range. A timer is only considered within the time range if
+// it is fully enclosed in it.
+struct TimeRange {
+  TimeRange(uint64_t start, uint64_t end)
+      : start(std::min(start, end)), end(std::max(start, end)) {}
```

Comment:
Hm. Why do we need such constructor? It might mask a bug in whatever instantiates `TimeRange`. I would suggest to just have an ORBIT_CHECK here instead.

Reply:
done